src/bot/scripts/card.js

```diff
@@ -728,946 +66,1233 @@ const card = new Command({
-    bg.crop(135, 165, 526, 769);
-
-    // save + post
-    const saveDate = Date.now();
-
-    bg.write(path.join(__dirname, "temp", `pd-${saveDate}.png`), async () => {
-
-      const embed = new discord.RichEmbed();
-
-      // card name
-      if (card.Name instanceof Array) {
-
-        embed.setAuthor(card.Name[0]);
-      } else {
-
-        embed.setAuthor(card.Name);
-      }
-
-      let embedColour = null;
-      switch (card.Theme) {
-      case "Adv":
-        embedColour = "#4f80ba";
-        break;
-      case "Fan":
-        embedColour = "#d34f5f";
-        break;
-      case "Sci":
-        embedColour = "#db571d";
-        break;
-      case "Mys":
-        embedColour = "#4b9b38";
-        break;
-      case "Sup":
-        embedColour = "#fd6cf8";
-        break;
-      case "Gen":
-        embedColour = "#857468";
-        break;
-      default:
-        embedColour = "#857468";
-      }
-      embed.setColor(embedColour);
-
-      embed.setDescription("");
-
-      embed.description += "**General Information**\n";
+        //checks if values are within bounds and some hardcoded shit
+        if (
+          (commandValues.modifier == "m" || commandValues.modifier === "l") &&
+          (commandValues.value > 7 || commandValues.value < 1)
+        ) {
+          return message.channel.send("Out of bounds");
+        }
 
-      embed.description += `Cast Area: ${card.CastArea}\n`;
+        if (
+          commandValues.modifier == "u" &&
+          (commandValues.value > 70 || commandValues.value < 1)
+        ) {
+          return message.channel.send("out of bounds");
+        }
+        if (commandValues.modifier === undefined) {
+          commandValues.modifier = "l";
+        }
+        if (commandValues.value === undefined) {
+          commandValues.value = 1;
+        }
 
-      if (card.CharacterType !== "Totem") {
+        if (commandValues.name === "%r %r") {
+          commandValues.name =
+            cards.data[Math.floor(Math.random() * cards.data.length)].name;
+          commandValues.modifier = "l";
+          commandValues.value = Math.floor(Math.random() * 7);
+        }
 
-        embed.description += `Max Speed: ${Math.round(card.MaxVelocity * 100) / 100}\n`;
-        embed.description += `Time To Reach Max Speed: ${Math.round(card.TimeToReachMaxVelocity * 100) / 100}\n`;
-        embed.description += `Agro Range Multiplier: ${Math.round(card.AgroRangeMultiplier * 100) / 100}\n\n`;
-      } else {
+        commandValues.value = parseInt(commandValues.value);
 
-        embed.description += "\n";
-      }
+        let highestToDate = 0;
+        let highestCard = null;
 
-      let hasPower = false;
-      for (let field in card) {
+        // card name - similarity
+        for (let card of cards.data) {
+          const lowerArray = card.name.toLowerCase();
 
-        if (field.startsWith("Power") && card[field] !== null) {
+          let sim = similarity(lowerArray, commandValues.name);
```

Comment:
youre not including the card aliases in the similarity search which is a current feature that a lot of people depend on, we cant ship with it missing, please re-read the gist i sent you

Reply:
do i just make another similarity? with the aliases

Reply:
just treat all the aliases as alternative names that can match as well

Reply:
I can only get the aliases through the id route, and not the list route, which will make me do a request on all the cards

Reply:
A workaround would be to add the aliasts in the list route.

Reply:
from the instructions i sent you; 'Note: the 'aliases' field will be added to https://sppd.feinwaru.com/api/v1/cards/list soon:tm:, you can expect it to be an array of strings (might be an empty [])'

either write the code and comment it out for now (since i havent added it yet, or see if the aliases field exists and if it doesnt, just run the code normally, but if it does, check the aliases for similarity as well)